### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by itouoti12** [AriPerkkio/vitest-sonar-reporter#12]
**Details**

I used this library to migrate from Jest to vitest.
I currently have 388 test files in my project and when I set up this library to create a reporter for SonarQube and ran vitest, I got the following error.

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Unhandled Error ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
RangeError: Maximum call stack size exceeded
 ❯ generateXml node_modules/vitest-sonar-reporter/dist/xml.js:58:12
 ❯ SonarReporter.onFinished node_modules/vitest-sonar-reporter/dist/sonar-reporter.js:37:35
 ❯ node_modules/vitest/dist/chunk-vite-node-externalize.6b27b039.mjs:10640:51
 ❯ Vitest.report node_modules/vitest/dist/chunk-vite-node-externalize.6b27b039.mjs:10638:38
 ❯ node_modules/vitest/dist/chunk-vite-node-externalize.6b27b039.mjs:10475:18
 ❯ processTicksAndRejections node:internal/process/task_queues:96:5
 ❯ Vitest.runFiles node_modules/vitest/dist/chunk-vite-node-externalize.6b27b039.mjs:10479:12
 ❯ Vitest.start node_modules/vitest/dist/chunk-vite-node-externalize.6b27b039.mjs:10406:5
 ❯ startVitest node_modules/vitest/dist/chunk-vite-node-externalize.6b27b039.mjs:11140:5
```

**Cause**
`src/xml.ts`

```ts
export function generateXml(files?: File[]) {
    return join(
        '<testExecutions version="1">',
        NEWLINE,
        ...(files?.map(generateFileElement).join(NEWLINE) || []),
        NEWLINE,
        '</testExecutions>'
    );
}
```

Spread Operator causes some timing to be broken down into an array of one character at a time

```
[
  ' ', ' ', '<', 'f', 'i', 'l', 'e',  ' ', 'p', 'a', 't',
  'h', '=', '"', 's', 'r', 'c', '/',  'p', 'a', 'g', 'e',
  's', 
  ... 491 more items
]
```

The spread operator (and Array.assign) loads the entire source array onto the stack, then pushes it onto the destination array.  Once the source array exceeds 125,052, it causes a Stack Overflow / Range Error.

**Suggestion**

```ts
export function generateXml(files?: File[]) {
    return join(
        '<testExecutions version="1">',
        NEWLINE,
        (files?.map(generateFileElement).join(NEWLINE) || []),
        NEWLINE,
        '</testExecutions>'
    );
}
```

Thanks for creating this wonderful library.

**Comment by AriPerkkio** [comment]
Thanks for the report @itouoti12! Your analysis seems correct.

```ts
files.map(generateFileElement).join(NEWLINE)
```

The `Array.join` returns a string which is then split into single characters by spread operator. This is unintended.

I believe a fix could be as simple as:
```diff
 return join(
     '<testExecutions version="1">',
     NEWLINE,
-    ...(files?.map(generateFileElement).join(NEWLINE) || []),
+    files?.map(generateFileElement).join(NEWLINE),
     NEWLINE,
     '</testExecutions>'
 );
```

Would you like to attempt to fix this and set up a pull request? Otherwise I'll take a look at this later this week.

**Comment by itouoti12** [comment]
@AriPerkkio 
Thanks for the confirmation. 

>Would you like to attempt to fix this and set up a pull request?
OK. I will create a pull request to resolve this issue.

**Comment by AriPerkkio** [comment]
This is now fixed and published in version `0.2.1`.
